components/places/src/storage.rs

```diff
@@ -139,36 +139,31 @@ pub fn apply_observation_direct(db: &Connection, visit_ob: VisitObservation) ->
 
     let mut update_frecency = false;
 
-    // There's a new visit, so update everything that implies
-    if let Some(visit_type) = visit_ob.visit_type {
-        // A single non-hidden visit makes the place non-hidden.
-        if !visit_ob.get_is_hidden() {
-            updates.push(("hidden", ":hidden", &false));
-        }
-        if visit_type == VisitTransition::Typed {
-            page_info.typed += 1;
-            updates.push(("typed", ":typed", &page_info.typed));
-        }
+    // There's a new visit, so update everything that implies. To help with
+    // testing we return the rowid of the visit we added.
+    let visit_row_id = match visit_ob.visit_type {
+        Some(visit_type) => {
+            // A single non-hidden visit makes the place non-hidden.
+            if !visit_ob.get_is_hidden() {
+                updates.push(("hidden", ":hidden", &false));
+            }
+            if visit_type == VisitTransition::Typed {
+                page_info.typed += 1;
```

Comment:
(This was already this way, but while you're here...) I think we probably should be setting update_frecency to true here?

Reply:
Aren't we still going to set it to true at line 159 - `if !visit_ob.is_error.unwrap_or(false)`?

Reply:
We do, but updating the `typed` field means that the frecency would be different if its recomputed, even if it were an error visit.

I guess I'm not really sure what the right thing to do is though...

Reply:
Ah, I see what you mean. Yeah, it's not completely clear, but I think what you suggest has more chance of being correct than what we have now, so I'll make the change.